CharityPortal/api/handlers/dashboard_handler.go

```diff
@@ -14,10 +13,12 @@ func NewDashboardHandler() *DashboardHandler {
 }
 
 func (ph *DashboardHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	userClaims := auth.GetUserFromContext(r.Context())
-	userData := &user.UserData{
-		ID:    userClaims.Sub,
-		Email: userClaims.Email,
-	}
-	views.Base(views.Dashboard(), userData).Render(r.Context(), w)
+	user := r.Context().Value(auth.UserContextKey{})
+    if user == nil {
+        http.Error(w, "User not in context", http.StatusUnauthorized)
+        return
+    }
```

Comment:
That wat the idea for session middleware

Reply:
Was it? The redirection here is not needed. Let me remove it, the requireLogin middle does it.